MaxText/max_utils.py

```diff
@@ -117,44 +115,13 @@ def upload_blob(destination_gcs_name, source_file_name):
   blob.upload_from_filename(source_file_name)
 
 def initialize_jax_distributed_system():
-  """ The jax distributed system is necessary for certain tasks such as asynchronous checkpointing in multihost settings.
-  Automatic arguments are chosen on cloud TPU starting with Jax 0.4.21
-  If are you in a different environment, e.g. GPUs, you will need to provide the appropriate arguments, see
-  https://jax.readthedocs.io/en/latest/_autosummary/jax.distributed.initialize.html
-  If you are unable to start the jax distributed system, you may remove the initialization attempt and instead use a
-  synchronous checkpointer (or no checkpointer) with async_checkpointing=False
-  (or enable_checkpointing=False to not use a checkpointer at all) """
-
-  def legacy_distribute_initialize():
-    """Calls jax.distribute.initialize() with appropriate multihost/multislice arguments.
-    This 'legacy' implementation uses the device backend (e.g. TPU backend), which
-    is forbidden starting in Jax version 0.4.21. The jax distributed system should be
-    initialized before the device backend, this is enforced starting with version 0.4.21."""
-
-    def gen_local_ip():
-      hostname = socket.gethostname()
-      return socket.gethostbyname(hostname)
-
-    def gen_local_ip_nums():
-      return [int(num) for num in gen_local_ip().split(':')[-1].split('.')]
-
-    def get_coordinator_ip():
-      local_ip_nums = jax.numpy.array(gen_local_ip_nums())
-      coordinator_ip_nums = multihost_utils.broadcast_one_to_all(local_ip_nums)
-      coordinator_ip_strings = [str(num) for num in list(coordinator_ip_nums)]
-      return '.'.join(coordinator_ip_strings)
-
-    port = multihost_utils.broadcast_one_to_all(jax.numpy.array(portpicker.pick_unused_port()))
-    coordinator_address = get_coordinator_ip() + ':' + str(port)
-    jax.distributed.initialize(coordinator_address=coordinator_address,
-                              num_processes=jax.process_count(),
-                              process_id=jax.process_index())
+  """ The best recipe to initialize the Jax Distributed System has varied over time. We keep a layer of
+      indirection in MaxText to avoid breaking the call sites unnecessarily.
 
+      Currently jax.distributed.initialize() fully works as expected!
+  """
   max_logging.log("Attempting to initialize the jax distributed system...")
-  if jax.__version__ >= '0.4.21':
```

Comment:
This change drops MaxText support for old versions of Jax - just confirming this is what we want?

Reply:
yes -- maxtext only supports ~HEAD.